model/src/cfd.rs

```diff
@@ -1403,10 +1379,45 @@ impl Cfd {
         }))
     }
 
+    fn event_with_reason(&self, event: EventKind, reason: anyhow::Error) -> CfdEvent {
+        self.log_cfd_event(&event, Some(reason));
+        self.issue_event_internal(event)
+    }
+
     fn event(&self, event: EventKind) -> CfdEvent {
+        self.log_cfd_event(&event, None);
+        self.issue_event_internal(event)
+    }
+
+    fn issue_event_internal(&self, event: EventKind) -> CfdEvent {
```

Comment:
not sure if this one is worth it; after all, it allows users of this API to bypass your mandatory logging introduced in this PR 🙃
I'd consider inlining these 2 calls.